comps/texttosql/langchain/main.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import pathlib
+import sys
+from typing import Annotated, Optional
+
+from pydantic import BaseModel, Field
+from sqlalchemy import create_engine
+from sqlalchemy.exc import SQLAlchemyError
+from src.texttosql import execute
+
+cur_path = pathlib.Path(__file__).parent.resolve()
+comps_path = os.path.join(cur_path, "../../../")
+sys.path.append(comps_path)
+
+from comps import opea_microservices, register_microservice
+
+
+class PostgresConnection(BaseModel):
+    user: Annotated[str, Field(min_length=1)]
+    password: Annotated[str, Field(min_length=1)]
+    host: Annotated[str, Field(min_length=1)]
+    port: Annotated[int, Field(ge=1, le=65535)]  # Default PostgreSQL port with constraints
+    database: Annotated[str, Field(min_length=1)]
+
+    def connection_string(self) -> str:
+        return f"postgresql://{self.user}:{self.password}@{self.host}:{self.port}/{self.database}"
+
+    def test_connection(self) -> bool:
+        """Test the connection to the PostgreSQL database."""
+        connection_string = self.connection_string()
+        try:
+            engine = create_engine(connection_string)
+            with engine.connect() as connection:
+                # If the connection is successful, return True
+                return True
+        except SQLAlchemyError as e:
+            print(f"Connection failed: {e}")
+            return False
+
+
+class Input(BaseModel):
+    input_text: str
+    conn_str: Optional[PostgresConnection] = None
+
+
+@register_microservice(
+    name="opea_service@texttosql",
+    endpoint="/v1/test-connection",
```

Comment:
using `/v1/postgres/health` to let user know it's a db connection check?

Reply:
Done